app/Espo/Core/Templates/Repositories/Hierarchy.php

```diff
@@ -398,6 +398,35 @@ public function getParentRecord(string $id): array
         return empty($record) ? [] : $record;
     }
 
+    protected function entityHasArchive(Entity $entity): bool
+    {
+        return !empty($this->getMetadata()->get(['scopes', $entity->getEntityType(), 'hasArchive']));
+    }
+
+
+    protected function validateIsArchived(Entity $entity): void
+    {
+        $fieldName = 'isArchived';
+        if ($entity->isAttributeChanged($fieldName) && $entity->get($fieldName)==true) {
+            // search all childs
+            if (!empty($entity->get('childrenIds'))) {
```

Comment:
There are no needs to find deeper than one level. Also, change $entity->get('childrenIds') to $children = $entity->get('children'), then check every item of collection and exit at first not archive value.

Reply:
Fixed